src/ArtView/ArtView.tsx

```diff
@@ -19,45 +33,53 @@ interface ImageDetails {
 }
 
 interface Props {
-  imageId: string;
-  artPieces: {
-    id: number;
-    image_src: string;
-    description: string;
-    title: string;
-    date_start: number;
-    image_id: string;
-    artist_title: string;
-    classification_titles: Array<string>;
-  }[]
+  category: string;
 }
 
 class ArtView extends Component<Props, State> {
   constructor(props: Props) {
     super(props)
     this.state = {
       userLikesPhoto: undefined,
-      singlePiece: undefined
+      singlePiece: undefined,
+      isLoading: true,
+      artPieces: [],
+      randomImageId: ''
     }
   }
 
-  findPiece = (value: boolean) => {
-    const foundPiece = this.props.artPieces.find(piece => piece.image_id === this.props.imageId)
-    this.setState({ userLikesPhoto: value, singlePiece: foundPiece })
+  componentDidMount = () => {
+    getArtDetails(this.props.category)
+      .then(data => {
+        this.setState({
+          isLoading: false,
+          artPieces: data.data,
+          randomImageId: data.data[this.getRandomIndex(data.data.length)].image_id
+        })
+      })
   }
 
+  getRandomIndex = (max: number) => {
+    return Math.floor(Math.random() * max);
+  }
 
+  findPiece = (value: boolean) => {
+    const foundPiece = this.state.artPieces.find(piece => piece.image_id === this.state.randomImageId)
+    this.setState({ userLikesPhoto: value, singlePiece: foundPiece })
+  }
+  
   render() {
+    const loading = this.state.isLoading ? <Loading /> : <><img className="featured-art" src={`https://www.artic.edu/iiif/2/${this.state.randomImageId}/full/843,/0/default.jpg`} alt="painting of people at a park on a sunny day" />
```

Comment:
Great if statement, we said that well make the else a component later

Reply:
Good plan!